owner/src/main/java/org/the_chance/honeymart/ui/orderdetails/OrderDetailsViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package org.the_chance.honeymart.ui.orderdetails
+
+import androidx.lifecycle.SavedStateHandle
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.update
+import org.the_chance.honeymart.domain.usecase.GetAllProductsByCategoryUseCase
+import org.the_chance.honeymart.domain.usecase.GetOrderDetailsUseCase
+import org.the_chance.honeymart.domain.usecase.GetOrderProductsDetailsUseCase
+import org.the_chance.honeymart.domain.util.ErrorHandler
+import org.the_chance.honeymart.ui.base.BaseViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class OrderDetailsViewModel @Inject constructor(
+    private val getOrderDetailsUseCase: GetOrderDetailsUseCase,
+    savedStateHandle: SavedStateHandle,
+) : BaseViewModel<OrderDetailsUiState, OrderDetailsUiEffect>(OrderDetailsUiState()){
+
+    override val TAG: String
+        get() = this::class.simpleName.toString()
+
+
+    init {
+        getData()
+    }
+
+    private fun getData() {
+        getOrderDetails(2)
+    }
+
+    private fun getOrderDetails(orderId: Long) {
+        _state.update { it.copy(isLoading = true, isError = false) }
+        tryToExecute(
+            { getOrderDetailsUseCase(orderId).toOrderParentDetailsUiState() },
```

Comment:
separate the part of a mapping in a separate function. you can put it in the function onGetOrederDetailsSuccess